I can't read autotools, so ditch all the garbage that is resulting
in a broken configure script in favor of a hardecoded sane check.

diff --git a/m4/curses.m4 b/m4/curses.m4
index 572d321..31db664 100644
--- a/m4/curses.m4
+++ b/m4/curses.m4
@@ -28,62 +28,8 @@ AC_DEFUN([IU_LIB_NCURSES], [
   AC_ARG_ENABLE(ncurses,    [  --disable-ncurses       don't prefer -lncurses over -lcurses],
               , enable_ncurses=yes)
   if test "$enable_ncurses" = yes; then
-    PKG_CHECK_MODULES([NCURSES], [ncursesw], [LIBNCURSES="${NCURSES_LIBS}" have_ncursesw=yes], [
-        PKG_CHECK_MODULES([NCURSES], [ncurses], [LIBNCURSES="${NCURSES_LIBS}" have_ncursesw=no], [
-            AC_CHECK_LIB(ncursesw, initscr, LIBNCURSES="-lncursesw",
-              AC_CHECK_LIB(ncurses, initscr, LIBNCURSES="-lncurses"))
-            if test "$ac_cv_lib_ncursesw_initscr" = yes; then
-              have_ncursesw=yes
-            else
-              have_ncursesw=no
-            fi
-        ])
-    ])
-    if test "$LIBNCURSES"; then
-      # Use ncurses header files instead of the ordinary ones, if possible;
-      # is there a better way of doing this, that avoids looking in specific
-      # directories?
-      AC_ARG_WITH([ncurses-include-dir],
-                  AS_HELP_STRING([--with-ncurses-include-dir=DIR],
-                         [Set directory containing the include files for
-                          use with -lncurses, when it isn't installed as
-                          the default curses library.  If DIR is "none",
-                          then no special ncurses include files are used.]))
-      if test "${with_ncurses_include_dir+set}" = set; then
-        AC_MSG_CHECKING(for ncurses include dir)
-	case "$with_ncurses_include_dir" in
-	  no|none)
-	    inetutils_cv_includedir_ncurses=none;;
-	  *)
-	    inetutils_cv_includedir_ncurses="$with_ncurses_include_dir";;
-	esac
-        AC_MSG_RESULT($inetutils_cv_includedir_ncurses)
-      else
-	AC_CACHE_CHECK(for ncurses include dir,
-		       inetutils_cv_includedir_ncurses,
-          if test "$have_ncursesw" = yes; then
-            ncursesdir=ncursesw
-          else
-            ncursesdir=ncurses
-          fi
-	  for D in $includedir $prefix/include /local/include /usr/local/include /include /usr/include; do
-	    if test -d $D/$ncursesdir; then
-	      inetutils_cv_includedir_ncurses="$D/$ncursesdir"
-	      break
-	    fi
-	    test "$inetutils_cv_includedir_ncurses" \
-	      || inetutils_cv_includedir_ncurses=none
-	  done)
-      fi
-      if test "$inetutils_cv_includedir_ncurses" = none; then
-        NCURSES_INCLUDE=""
-      else
-        NCURSES_INCLUDE="-I$inetutils_cv_includedir_ncurses"
-      fi
-    fi
-    if test $have_ncursesw = yes; then
-      AC_DEFINE(HAVE_NCURSESW, 1, [Define if you have working ncursesw])
-    fi
+    PKG_CHECK_MODULES([NCURSES], [ncursesw], [LIBNCURSES="${NCURSES_LIBS}" have_ncursesw=yes])
+    AC_DEFINE(HAVE_NCURSESW, 1, [Define if you have working ncursesw])
   fi
   AC_SUBST(NCURSES_INCLUDE)
   AC_SUBST(LIBNCURSES)])dnl
